app/decorators/backend_api_decorator.rb

```diff
@@ -17,6 +17,12 @@ def add_backend_usage_backends_data
     }
   end
 
+  def products_used_table_data
+    ServiceDecorator.decorate_collection(services.accessible.sort_by(&:name))
```

Comment:
I liked the `order` version though. Why the change?

Reply:
Another remark, customers with backend api shared across many services might have hard time to load this ...
Do you know about any UI/UX solution @cntlsn @josemigallas @3scale/system-ux  ?

Reply:
@hallelujah can you please expand on why it could take longer to load the list? one of the main reasons behind this UI change is actually to try to speed up page loading.

Reply:
Well we have issues already with customers with more than 500 services on-prem ...
I imagine that this will generate a huge JSON and also a huge data attribute.

Reply:
@hallelujah I couldn't make `order` work.

As per performance, the JSON is not an issue, the bottleneck would be the DB I think. But no I have not tested it.

Reply:
I'd say the problem will be it might be slow :) Do not think only if it is DB or JS
Probably the implementation is not solving at all the problem as quoted and stressed below:

> @hallelujah can you please expand on why it could take longer to load the list? **one of the main reasons behind this UI change is actually to try to speed up page loading.**

JS/ DB ruby are only tools but if you load all the services there is no miracle ... it will be slow :(

I'd propose an enhancement loading the services partially (20 services at a time for example, paginating) and the search would be done in AJAX. _This is a proposition of implementation not a specification_

Reply:
Yes, loading items per-page is basically what we are looking for in the new design. But it makes the implementation much harder and IMO moving to react and cleaning up old logic will be an improvement. Actually I think we can't really assure anything until we start stress testing.

Reply:
Did you do a test locally creating the 1k services and accounts with FactoryBot?
By the way what failed with the order query? Can I help you with that?
It is better to do it from the Database as it will be faster